### DISCUSSION TITLE
How is the coverage functions of the template are to be used?

### DISCUSSION THREAD

**Discussion by hpe-ykoehler** [friendlyanon/cmake-init#84] [General]
Is there some instructions on how the coverage aspect of the template are to be used?

I ran cmake build with preset dev-coverage and target all, I see my code compiling, test run, then I run cmake with target coverage, I get a coverage.info file but it is empty, I am unclear what I am doing wrong.

**Comment by friendlyanon** [comment]
There is a preset that is run in CI on Ubuntu using Clang to get you started with coverage. There used to be an OpenCppCoverage file as well as a template to help people do coverage on Windows as well, but I deemed it unnecessary. The file [cmake-init/templates/common/cmake/coverage.cmake](https://github.com/friendlyanon/cmake-init/blob/master/cmake-init/templates/common/cmake/coverage.cmake) contains some targets for lcov and the necessary flags for coverage instrumentation are passed in via [presets](https://github.com/friendlyanon/cmake-init/blob/master/cmake-init/templates/common/CMakePresets.json#L117-L138). The `dev-coverage` user preset basically just forwards to this preset.

For complete coverage instructions, you may just refer to the [coverage](https://github.com/friendlyanon/cmake-init/blob/master/cmake-init/templates/common/.github/workflows/ci.yml#L35) CI job.

**Comment by hpe-ykoehler** [comment]
ok, I was trying to figure out how I could use that in vscode, the cmake integration is not as clear as running the CLI commands...  I will try to figure this out based on the CLI commands seen in there, thanks.
